File: trunk/test/scala/org/codehaus/testdox/intellij/ui/AddTestDialogTest.scala
package org.codehaus.testdox.intellij.ui

import org.intellij.openapi.testing.DialogCreator
import junit.framework.Assert.assertEquals

class AddTestDialogTest extends RenameDialogTest {

  def testHasADifferentTitleThanTheRenameDialog() {
    assertEquals("Add Test", createDialog().getTitle())
  }

  def testStuff() {
    ("foo bar baz".replaceAll("\\s", ""): Seq[Char]) match {
      case Seq(first, rest@_*) => println("first: " + first + ", rest: " + rest)
      case Seq(_*) => println("nada")
    }
  }

  override protected def createDialogCreator() = new DialogCreator[RenameDialog]() {
    def create(): RenameDialog = new AddTestDialog(projectMock).asInstanceOf[RenameDialog]
  }
}

File: branches/pre-scala-migration/src/test/org/codehaus/testdox/intellij/panel/AddTestDialogTest.java
package org.codehaus.testdox.intellij.panel;

import org.intellij.openapi.testing.DialogCreator;

public class AddTestDialogTest extends RenameDialogTest {

    public void testHasADifferentTitleThanTheRenameDialog() throws Exception {
        assertEquals("Add Test", createDialog().getTitle());
    }

    protected DialogCreator<AddTestDialog> dialogCreator() {
        return new DialogCreator<AddTestDialog>() {
            protected AddTestDialog create() {
                return new AddTestDialog(projectMock);
            }
        };
    }
}
